src/test/objectstore/store_test.cc

```diff
@@ -3414,6 +3414,215 @@ TEST_P(StoreTest, OmapSimple) {
   }
 }
 
+TEST_P(StoreTest, OmapBench) {
+  // Random dictionary generation
+  static const std::string dict =
+      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+  mt19937 gen{random_device{}()};
+  uniform_int_distribution<> dist(0, dict.size() - 1);
+
+  // options
+  int obj_count = 100000; // number of objects
+  int keys_per_object = 100; // number of keys per object
+  int key_batch_size = 100; // number of keys to batch at once
+  int klen = 64; // omap key size in bytes
+  int vlen = 256; // omap value size in bytes
+  int hash_count = 100;  // number of unique hashes to use for hobject_t hash. (simulates PGs)
+  bool obj_sort = true; // sort the ghobject_t vector using std::sort (helps rosckdb avoid random io for example).
+
+  int r;
+  vector<ghobject_t> objects;
+  float min = std::numeric_limits<float>::max();
+  float max = 0, avg = 0, total = 0;
+
+  // Make the collection
+  coll_t cid;
+  auto ch = store->create_new_collection(cid);
+  {
+    ObjectStore::Transaction t;
+    t.create_collection(cid, 0);
+    cerr << "Creating collection " << cid << std::endl;
+    r = queue_transaction(store, ch, std::move(t));
+    ASSERT_EQ(r, 0);
+  }
+
+  cout << "[OMAPBENCH] *** Setting up objects ***" << std::endl;
+  std::unordered_map<size_t, size_t> hashes;
+
+  for (int obj = 0; obj < obj_count; obj++) {
+    string name("object");
+    name += stringify(obj);
+
+    // Make a hash to simulate PGs
+    auto hhash = std::hash<string>{}(name) % hash_count;
+
+    objects.push_back(ghobject_t(hobject_t(sobject_t(name, CEPH_NOSNAP),
+                                "key", hhash, -1, "")));
+    auto hash = std::hash<ghobject_t>{}(objects.back());
+    hashes[hash]+=1;
+  }
+  cout << "[OMAPBENCH] objects: " << obj_count << ", unique object hashes: " << hashes.size() << std::endl;
+
+  // Sort the objects?
+  if (obj_sort) {
+    std::sort(objects.begin(), objects.end(), [](ghobject_t const& a, ghobject_t const& b) { return a < b;});
```

Comment:
No need to supply custom compare operator, operator<(ghobject_t, ghobject_t) is used when not provided.

Reply:
hrm, I seem to remember it actually didn't work properly without the custom compare operation, but I don't remember why now.  I'll take a look.

Reply:
Works fine now anyway, so easy fix!